compiler/src/dotty/tools/dotc/transform/PublicInBinaryAnnotations.scala

```diff
@@ -0,0 +1,35 @@
+package dotty.tools.dotc.transform
+
+import dotty.tools.dotc.core.Contexts.*
+import dotty.tools.dotc.core.DenotTransformers.SymTransformer
+import dotty.tools.dotc.core.Flags.*
+import dotty.tools.dotc.core.Symbols.NoSymbol
+import dotty.tools.dotc.core.SymDenotations.SymDenotation
+import dotty.tools.dotc.transform.MegaPhase.MiniPhase
+import dotty.tools.dotc.typer.RefChecks
+
+/** Makes @publicInBinary definitions public.
+ *
+ *  This makes it possible to elide the generations of some unnecessary accessors.
+ */
+class PublicInBinary extends MiniPhase with SymTransformer:
+
+  override def runsAfterGroupsOf: Set[String] = Set(RefChecks.name)
+
+  override def phaseName: String = PublicInBinary.name
+  override def description: String = PublicInBinary.description
+
+  def transformSym(d: SymDenotation)(using Context): SymDenotation = {
+    if d.hasPublicInBinary then
+      d.resetFlag(Protected)
```

Comment:
@sjrd points out that `protected def` actually always becomes public in bytecode (we never emit ACC_PROTECTED), so ProtectedAccessors generates unnecessary accessors when the method doesn't come from Java (but we can't change its behavior without affecting binary compat)

Reply:
I removed this transformation phase. We do not need to update the backend as all the members marked with `@publicInBinary` already become public in the bytecode. I also added a simple condition in `ProtectedAccessors` to avoid the creation of the accessors for protected members marked as `@publicInBinary`.